src/test/java/core/basesyntax/strategy/OperationStrategyTest.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.strategy;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.dao.ProductDao;
+import core.basesyntax.dao.impl.ProductDaoImpl;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitTransaction.Operation;
+import core.basesyntax.strategy.impl.BalanceOperationServiceImpl;
+import core.basesyntax.strategy.impl.PurchaseOperationServiceImpl;
+import core.basesyntax.strategy.impl.ReturnOperationServiceImpl;
+import core.basesyntax.strategy.impl.SupplyOperationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class OperationStrategyTest {
+    private static ProductDao productDao;
+    private static OperationStrategy operationStrategy;
+
+    @BeforeAll
+    static void setUp() {
+        productDao = new ProductDaoImpl();
+        operationStrategy = new OperationStrategy(productDao);
+    }
+
+    @Test
+    void getOperation_getBalanceOperation_ok() {
+        Class<BalanceOperationServiceImpl> expected = BalanceOperationServiceImpl.class;
+        OperationService actual = operationStrategy.getOperation(
+                new FruitTransaction(Operation.BALANCE, "banana", 10));
+        assertEquals(expected, actual.getClass());
+    }
+
+    @Test
+    void getOperation_getReturnOperation_ok() {
+        OperationService expected = new ReturnOperationServiceImpl(productDao);
```

Comment:
please fix all test methods in this class in the same way as method:
```
void getOperation_getBalanceOperation_ok() {
        Class<BalanceOperationServiceImpl> expected = BalanceOperationServiceImpl.class;
        OperationService actual = operationStrategy.getOperation(
                new FruitTransaction(Operation.BALANCE, "banana", 10));
        assertEquals(expected, actual.getClass());
    }
```
Remove `productDao` as class field, it will be redundant.